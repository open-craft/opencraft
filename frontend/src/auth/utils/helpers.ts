import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
