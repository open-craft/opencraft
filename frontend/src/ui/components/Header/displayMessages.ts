const messages = {
  faq: {
    defaultMessage: 'FAQ',
    description: 'Navbar link for FAQ page'
  },
  login: {
    defaultMessage: 'Login',
    description: 'Navbar link for Login page'
  },
  registration: {
    defaultMessage: 'Create your account',
    description: 'Navbar link for Registration page'
  },
  customize: {
    defaultMessage: 'Customize',
    description: 'Navbar link for Customization / Console page'
  },
  visitLMS: {
    defaultMessage: 'My site (LMS)',
    description: 'Navbar link for LMS'
  },
  siteSettings: {
    defaultMessage: 'Site Settings',
    description: 'Navbar link for Site Settings page'
  },
  account: {
    defaultMessage: 'Account',
    description: 'Navbar link for User Account page'
  },
  domain: {
    defaultMessage: 'Domain',
    description: 'Navbar link for Domain management page'
  },
  noticeBoard: {
    defaultMessage: 'Status & Notifications',
    description: 'Navbar link for Noticeboard page'
  },
  logout: {
    defaultMessage: 'Logout',
    description: 'Navbar link for Logout'
  },
  goToConsole: {
    defaultMessage: 'Go to console',
    description: 'Console is available when logged in'
  }
};

export default messages;
