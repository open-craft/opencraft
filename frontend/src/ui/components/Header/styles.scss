@import '~styles/theme';


.logo-container {
  display: flex;
  justify-content: center;

  .site-logo {
    fill: #fff;
    width: 152px;
    height: 82px;
    padding-top: 15px;
  }
}

/* change the link color */
.navbar-dark {
  .navbar-toggler {
    border: none;
  }

  .navbar-nav .nav-link {
    font-family: $chivo-font;
    font-size: 20px;
    letter-spacing: 0.72px;
    color: white;
    border-bottom: 3px solid transparent;
    padding-left: none;
    padding-right: none;
    margin: auto 7px;

    &:hover, &:focus {
      color: white;
      border-bottom: 3px solid white;
    }
  }
}
