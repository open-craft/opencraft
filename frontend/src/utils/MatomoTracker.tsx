import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
