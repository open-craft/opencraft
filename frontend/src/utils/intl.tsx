import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
