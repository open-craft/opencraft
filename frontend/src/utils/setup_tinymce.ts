import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
