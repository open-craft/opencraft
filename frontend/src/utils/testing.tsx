import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
