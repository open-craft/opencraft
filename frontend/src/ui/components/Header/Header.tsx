import * as React from 'react';
import { Navbar, Nav, Container } from 'react-bootstrap';
import { ROUTES, CONTACT_US_LINK } from 'global/constants';
import { useLocation, useHistory } from 'react-router-dom';
import logo from 'assets/icons.svg';
import './styles.scss';

export const Header: React.FC = () => {
  const currentLocation = useLocation().pathname;
  const history = useHistory();

  // Workaround to check authentication on header instead of connecting
  // component to redux
  const authenticated = currentLocation.includes(ROUTES.Console.HOME);

  if (authenticated) {
    return (
      <Container>
        <Navbar bg="transparent" expand="lg" variant="dark">
          <Navbar.Brand className="logo-container mx-auto">
            <svg className="site-logo">
              <use xlinkHref={`${logo}#opencraft_logo`} />
            </svg>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ml-auto">
              <Nav.Link>Customize</Nav.Link>
              <Nav.Link onClick={() => window.open(CONTACT_US_LINK, '_blank')}>
                Support Request
              </Nav.Link>
              <Nav.Link disabled>Status & Notifications</Nav.Link>
              <Nav.Link
                onClick={() => {
                  history.push(ROUTES.Auth.LOGOUT);
                }}
              >
                Log out
              </Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      </Container>
    );
  }

  // Only show login link when on registration page and only show registration
  // link when on login page
  return (
    <Container>
      <Navbar bg="transparent" expand="md" variant="dark">
        <Navbar.Brand className="logo-container mx-auto">
          <svg className="site-logo">
            <use xlinkHref={`${logo}#opencraft_logo`} />
          </svg>
        </Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="ml-auto">
            {currentLocation.includes(ROUTES.Registration.HOME) && (
              <Nav.Link onClick={() => history.push(ROUTES.Auth.LOGIN)}>
                Login
              </Nav.Link>
            )}
            {currentLocation.includes(ROUTES.Auth.LOGIN) && (
              <Nav.Link onClick={() => history.push(ROUTES.Registration.HOME)}>
                Create your account
              </Nav.Link>
            )}
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </Container>
  );
};
