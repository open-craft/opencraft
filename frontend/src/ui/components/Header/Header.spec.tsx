import React from 'react';
import { setupComponentForTesting } from 'utils/testing';
import { Header } from './Header';

it('renders without crashing', () => {
  const tree = setupComponentForTesting(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
