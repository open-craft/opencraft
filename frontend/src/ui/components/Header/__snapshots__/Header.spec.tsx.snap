// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="container"
>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-transparent"
  >
    <span
      className="logo-container mx-auto navbar-brand"
    >
      <svg
        className="site-logo"
      >
        <use
          xlinkHref="icons.svg#opencraft_logo"
        />
      </svg>
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      />
    </div>
  </nav>
</div>
`;
